### ISSUE TITLE
pg_dump: warning: there are circular foreign-key constraints on this table

### ISSUE THREAD

**Issue by ventsislavnikolov** [railwayapp-templates/postgres-s3-backups#12]
Hello, 2 day ago I made everything by the tutorial but it gives me this error and doesnt create backup:

`containers-us-west-22.railway.app:8007 - accepting connections

pg_dump (PostgreSQL) 16.1

NodeJS Version: v18.18.2

Backup cron scheduled...

Initiating DB backup...

Dumping DB to file...

Error while running backup:  {

stderr: 'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: hypertable\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.\n' +

'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: chunk\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.'

}

Initiating DB backup...

Dumping DB to file...

Error while running backup:  {

stderr: 'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: hypertable\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.\n' +

'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: chunk\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.'

}`

I have env variables: 
![image](https://github.com/railwayapp-templates/postgres-s3-backups/assets/34811964/b5fc2a57-5510-47d0-9645-cba308637398)

Did I miss something? Can somebody help?

Thanks!

**Comment by mbrookson** [comment]
Just came to report the same issue.

Seeing this every hour and backups are no longer working. I think it's started happening since upgrading my Postgres instance from the legacy plugin architecture to the new Railway database with volume setup. Not sure what's changed under the hood to cause it.


```sh
Error while running backup:  {

stderr: 'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: hypertable\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.\n' +

'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: chunk\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.\n' +

'pg_dump: warning: there are circular foreign-key constraints on this table:\n' +

'pg_dump: detail: continuous_agg\n' +

'pg_dump: hint: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.\n' +

'pg_dump: hint: Consider using a full dump instead of a --data-only dump to avoid this problem.'

}
```

![image](https://github.com/railwayapp-templates/postgres-s3-backups/assets/20131783/ec178bd9-ca02-4032-a942-7599eb6c97da)

**Comment by ventsislavnikolov** [comment]
I think I found my problem, my data is corrupted or have some problems between relations on Strapi. I think this still make some backup but dunno if backup everything.

**Comment by rkm3** [comment]
Same problem here. Will try @RyanElder's suggestion.

**Comment by brody192** [comment]
fixed in #13
